### PR TITLE
Add support for --branch argument

### PR DESCRIPTION
**Pull request by dayllanmaza**
Bug: T266154

### CODE REVIEW

File: bin/deploy.sh [hunk 1]
```diff
@@ -17,23 +23,23 @@ git fetch --quiet origin 2>&1
 HIGHEST_TAG=$(git tag --list "*.*.*" | sort --version-sort | tail --lines 1)
 CURRENT_TAG=$(git describe --tags --always)
 CURRENT_BRANCH=$(git symbolic-ref --short -q HEAD)
-DIFF_TO_MASTER=$(git diff origin/master)
+DIFF_TO_BRANCH=$(git diff origin/$BRANCH)
```
Comment (samwilson) [inline bin/deploy.sh:31]: I'm not sure if it matters, but if the given branch doesn't exist then this shows a fatal error. Should we check for its existence first (with e.g. `git ls-remote origin $BRANCH`)?

Comment (dayllanmaza) [reply bin/deploy.sh:31]: I think it is a good idea. I'll add that.

File: bin/deploy.sh [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+    if [[ -z $(git ls-remote origin $2) ]]; then
+        echo "$2 branch does not exist. Not deploying."
+        exit 0
+    else
+        BRANCH="$2"
+    fi
+fi
 
 ## Update the repo.
 cd $APP_DIR
```
Comment (samwilson) [inline bin/deploy.sh:24]: This needs to have been higher up. Will make a patch.

Comment (samwilson) [reply bin/deploy.sh:24]: https://github.com/wikimedia/ToolforgeBundle/pull/37

**Pull request merged**